src/sdl/sdl_display.c

```diff
@@ -144,14 +144,27 @@ static ALLEGRO_DISPLAY *sdl_create_display_locked(int w, int h)
       ALLEGRO_ERROR("SDL_CreateWindow failed: %s", SDL_GetError());
       return NULL;
    }
-   SDL_GetWindowSize(sdl->window, &d->w, &d->h);
 
    flags =
       SDL_RENDERER_ACCELERATED |
       SDL_RENDERER_PRESENTVSYNC |
       SDL_RENDERER_TARGETTEXTURE;
    sdl->renderer = SDL_CreateRenderer(sdl->window, -1, flags);
    sdl->context = SDL_GL_GetCurrentContext();
+
+   SDL_GetRendererOutputSize(sdl->renderer, &d->w, &d->h);
+
+   // there's no way to query pixel ratio before creating the window, so we
+   // have to compensate afterwards
+   if (d->flags & ALLEGRO_RESIZABLE &&
+        !(d->flags & ALLEGRO_FULLSCREEN || d->flags & ALLEGRO_FULLSCREEN_WINDOW)) {
+      int window_width, window_height;
```

Comment:
Would you mind adding an ALLEGRO_DEBUG saying that it's resizing, and also ALLEGRO_WARN if it wasn't resizable but the ration is not 1.?

Reply:
If it wasn't resizable, then the hidpi support wouldn't be enabled at all (most likely resulting in automatic resize by system compositor with Allegro/SDL not even being aware of that): https://github.com/liballeg/allegro5/pull/999/files#diff-49d564f1704de963274032237e706cc9R105-R106

If we want to allow hidpi support on non-resizable Allegro windows, I guess a new API would have to be created on Allegro side to handle that.